src/Npgsql/Internal/NpgsqlReadBuffer.cs

```diff
@@ -111,8 +113,10 @@ internal TimeSpan Timeout
             Underlying = stream;
             _underlyingSocket = socket;
             Cts = new ResettableCancellationTokenSource();
-            Size = size;
-            Buffer = ArrayPool<byte>.Shared.Rent(size);
+            Buffer = usePool ? ArrayPool<byte>.Shared.Rent(size) : new byte[size];
```

Comment:
At this point it seems simpler to just allocate the buffer outside of NpgsqlReadBuffer and pass it in? For the return side, the connector knows when a buffer is oversize or not, so it should know when it needs to be returned to the ArrayPool... Am I missing some complication?

Reply:
Indeed it would be possible but I do not see how it could be simpler...
The connector class has already a lot of business logic to handle, I would let the read/write buffer classes handling their own buffer allocation logic.
Even more, I would let the `NpgsqlReadBuffer` class fully handling the oversize allocation when needed, but maybe I miss some complication on that part also...

Reply:
OK, it isn't too important one way or the other.